src/k6StudioState.ts

```diff
@@ -0,0 +1,22 @@
+import { Process } from '@puppeteer/browsers'
+import { ChildProcessWithoutNullStreams } from 'child_process'
+import eventEmitter from 'events'
+
+import { ProxyStatus } from './types'
+
+export type k6StudioState = {
+  currentBrowserProcess: Process | ChildProcessWithoutNullStreams | null
+  proxyStatus: ProxyStatus
+  proxyEmitter: eventEmitter
+}
+
+export function initialize() {
+  globalThis.k6StudioState = {
```

Comment:
This is where the global variables are initialized

Reply:
I love the idea of injecting the dependencies in the handlers' initialization to avoid the global variable. I've made a ticket and linked your comment to it so we can address it once the initial refactoring is complete. https://github.com/grafana/k6-studio/issues/634